pkg/assets/interface.go

```diff
@@ -26,4 +26,5 @@ type StoreGetter interface {
 	GetSecretOrConfigMapKey(key monitoringv1.SecretOrConfigMap) (string, error)
 	GetConfigMapKey(key v1.ConfigMapKeySelector) (string, error)
 	GetSecretKey(key v1.SecretKeySelector) ([]byte, error)
+	GetNamespace() string
```

Comment:
(suggestion) we could make TLSAsset() part of the StoreGetter interface

TLSAsset(key interface{}) string
```suggestion
	TLSAsset(key interface{}) string
```

Reply:
Hmmm.... Maybe make`GetNamespace() string` part of the StoreGetter interface is good.
When `s := store.ForNamespace(ns)`, StoreGetter can not get namespace.
```golang
// ForNamespace returns a StoreGetter scoped to the given namespace.
// It reads data only from the cache which needs to be populated beforehand.
// The namespace argument can't be empty.
func (s *StoreBuilder) ForNamespace(namespace string) StoreGetter {
	if namespace == "" {
		panic("namespace can't be empty")
	}
	return &cacheOnlyStore{
		ns: namespace,
		c:  s.objStore,
	}
}

type cacheOnlyStore struct {
	ns string
	c  cache.Store
}
```

Reply:
My idea was to adapt the existing TLSAsset() function like this:

```golang
func (cos *cacheOnlyStore) TLSAsset(sel interface{}) string {
    var k tlsAssetKey

    switch v := sel.(type) {
    case monitoringv1.SecretOrConfigMap:
        k = tlsAssetKeyFromSelector(cos.ns, v)
    case *v1.SecretKeySelector:
        k = tlsAssetKeyFromSecretSelector(cos.ns, v)
    default:
        return ""
    }

    return k.toString()
}

```

Reply:
I got it. Thanks.